core/templates/dev/head/pages/exploration_editor/editor_tab/SolutionValidityService.js

```diff
@@ -0,0 +1,24 @@
+// Service for keeping track of solution validity.
+oppia.factory('SolutionValidityService', [
```

Comment:
Can you explain what kind of tests I should write for this -- as in should I be writing frontend tests (spec files) or e2e tests? I tried writing frontend tests but it was way too confusing. I'm working on it.